### PR TITLE
we must add CreateChatInviteLink and some assistants and change name one method

### PR DESCRIPTION
**Pull request by ABOLFAZLVALIKHANI**
I added the `CreateChatInviteLink `method. For this, I added the assistants folder to the project.
We can use assistants later , for example:

- Better access to return values ​​from the Telegram side
- Adding some general functionality, for example checking a user's membership

Assistants can be used in various ways. I will try to add more general features to this section in due course

change name method `kickChatMember `to `banChatMember`
I kept the previous name so that users don't have problems when changing the version

### CODE REVIEW

**Comment by Ali-Toosi** [comment]
Thanks, I'll review this tomorrow.

**Comment by ABOLFAZLVALIKHANI** [comment]
you're welcome

**Comment by ABOLFAZLVALIKHANI** [comment]
you're welcome

File: django_tgbot/bot_api_user.py [hunk 1]
```diff
@@ -125,12 +129,17 @@ def request_and_result(self, data, result_type, files=None):
         """
         res = self.send_request(inspect.stack()[1].function, data=data, files=files)
         if res['ok']:
+            # print(res['result'])
+
```
Comment (Ali-Toosi) [inline django_tgbot/bot_api_user.py]:
```suggestion
```

File: django_tgbot/bot_api_user.py [hunk 2]
```diff
@@ -125,12 +129,17 @@ def request_and_result(self, data, result_type, files=None):
...
         res = self.send_request(inspect.stack()[1].function, data=data, files=files)
         if res['ok']:
+            # print(res['result'])
+            
             if type(result_type) == list and len(result_type) > 0:
                 if len(result_type) > 1:
                     raise ValueError("Passed `result_type` cannot have more than one element if it is a list.")
                 return list(map(result_type[0], list(res['result'])))
             else:
+
```
Comment (Ali-Toosi) [inline django_tgbot/bot_api_user.py]:
```suggestion
```

File: django_tgbot/bot_api_user.py [hunk 3]
```diff
@@ -125,12 +129,17 @@ def request_and_result(self, data, result_type, files=None):
...
+            
             if type(result_type) == list and len(result_type) > 0:
                 if len(result_type) > 1:
                     raise ValueError("Passed `result_type` cannot have more than one element if it is a list.")
                 return list(map(result_type[0], list(res['result'])))
             else:
+                
                 return result_type(res['result'])
+                
+
```
Comment (Ali-Toosi) [inline django_tgbot/bot_api_user.py]:
```suggestion
```

File: django_tgbot/bot_api_user.py [hunk 4]
```diff
@@ -359,6 +384,10 @@ def setChatPermissions(self, chat_id, permissions):
 
     def exportChatInviteLink(self, chat_id):
         return self.request_and_result(create_params_from_args(locals()), bool)
+    
+    def createChatInviteLink(self , chat_id , name : str = None , expire_date : int = None , member_limit : int = 5 , 
+                            creates_join_request : bool  = False ) :
+        return self.request_and_result(create_params_from_args(locals()), CreateChatInviteLink )
```
Comment (Ali-Toosi) [inline django_tgbot/bot_api_user.py]:
```suggestion
    
    def createChatInviteLink(self, chat_id, name : str = None, expire_date : int = None, member_limit : int = 5, 
                            creates_join_request : bool  = False) :
        return self.request_and_result(create_params_from_args(locals()), CreateChatInviteLink)
```

Comment (Ali-Toosi) [reply django_tgbot/bot_api_user.py]: In addition to removing the spaces as above, please also make sure the parameters are exactly like they are defined in Telegram API. In their API, there is no default value for `member_limit` or `creates_join_request`.

File: django_tgbot/assistants/createChatInviteLink.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import inspect
```
Comment (Ali-Toosi) [inline django_tgbot/assistants/createChatInviteLink.py]: Not used, can be deleted.

```suggestion
```

File: django_tgbot/assistants/createChatInviteLink.py
Comment (Ali-Toosi) [inline django_tgbot/assistants/createChatInviteLink.py:1]: I understand you want to use this as the type for a `ChatInviteLink`. However, there already is a folder for all the types. Please move this to `django_tgbot/types/chatinvitelink.py` and make sure its style is the same as other types in that folder (should inherit the same class, the class name should be exactly like Telegram `ChatInviteLink`,  etc.).

**Review by Ali-Toosi — CHANGES_REQUESTED** [review]

File: django_tgbot/assistants/createChatInviteLink.py
Comment (ABOLFAZLVALIKHANI) [reply django_tgbot/assistants/createChatInviteLink.py:1]: `ChatInviteLink` is not a TYPE [(!)](https://core.telegram.org/bots/api#createchatinvitelink), Therefore, I suggest not to put it in the type folder.
I already tried to take this as Type. But it was not ideal and I encountered some problems.
so we can't  make sure its style is the same as other types!

Comment (Ali-Toosi) [reply django_tgbot/assistants/createChatInviteLink.py:1]: It is a type: https://core.telegram.org/bots/api#chatinvitelink
You are referring to the method. But there is also a type. That method basically creates one ChatInviteLink object.

Comment (Ali-Toosi) [reply django_tgbot/assistants/createChatInviteLink.py:1]: I'm closing the PR. Feel free to re-open it if you wanted to make changes.

**Pull request closed without merging**